controllers/volumereplication_controller.go

```diff
@@ -239,3 +270,43 @@ func (r *VolumeReplicationReconciler) resyncVolume(volumeID string, parameters,
 
 	return tasks.RunAll(resyncVolumeTasks)
 }
+
+// disableVolumeReplication defines and runs a set of tasks required to disable volume replication
+func (r *VolumeReplicationReconciler) disableVolumeReplication(volumeID string, parameters, secrets map[string]string) (string, error) {
+	c := replication.CommonRequestParameters{
+		VolumeID:    volumeID,
+		Parameters:  parameters,
+		Secrets:     secrets,
+		Replication: r.Replication,
+	}
+
+	var disableVolumeReplicationTasks = []*tasks.TaskSpec{
+		{
+			Name: disableVolumeReplication,
+			Task: replication.NewDisableTask(c),
+		},
+	}
+
+	return tasks.RunAll(disableVolumeReplicationTasks)
+}
+
+// Checks whether a string is contained within a slice
+func contains(slice []string, s string) bool {
+	for _, item := range slice {
+		if item == s {
+			return true
+		}
+	}
+	return false
+}
+
+// Removes a given string from a slice and returns the new slice
+func remove(slice []string, s string) (result []string) {
+	for _, item := range slice {
+		if item == s {
+			continue
+		}
+		result = append(result, item)
+	}
+	return
+}
```

Comment:
These are common utility functions. Keep it in a separate `controllers/strings.go` file?